http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/Queries.scala

```diff
@@ -0,0 +1,146 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package db
+
+import java.time.Instant
+
+import cats.data.Chain
+import org.broadinstitute.dsde.workbench.leonardo.db.LeoProfile.api._
+import org.broadinstitute.dsde.workbench.leonardo.db.LeoProfile.mappedColumnImplicits._
+import org.broadinstitute.dsde.workbench.leonardo.http.service.ListClusterResponse
+import org.broadinstitute.dsde.workbench.leonardo.model.{Cluster, DataprocInfo}
+import org.broadinstitute.dsde.workbench.leonardo.model.google.{ClusterStatus, IP, OperationName}
+import org.broadinstitute.dsde.workbench.model.google.{GcsBucketName, GoogleProject}
+import cats.implicits._
+import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
+
+import scala.concurrent.ExecutionContext
+
+object Queries {
+  type ClusterJoinLabelJoinRuntimeConfig = Query[((ClusterTable, Rep[Option[LabelTable]]), Rep[Option[RuntimeConfigTable]]), ((ClusterRecord, Option[LabelRecord]), Option[RuntimeConfigRecord]), Seq]
+
+  val runtimeConfigs = TableQuery[RuntimeConfigTable]
+
+  val clusterLabelRuntimeConfigQuery: ClusterJoinLabelJoinRuntimeConfig = {
+    for {
+      (cluster, label) <- clusterQuery
```

Comment:
There's a subtle issue here. If you filter on this query, e.g.:
```
clusterLabelRuntimeConfigQuery.filter(_._1 === "foo").filter(_._._2 === "bar").result
```
the SQL expands to something horrendously inefficient, resulting in full table scans in MySQL even though all fields are properly indexed.

The solution is to do the filtering _before_ the joins, e.g.:
```
def clusterLabelRuntimeConfigQuery(baseClusterQuery: Query[...]): ClusterJoinLabelJoinRuntimeConfig = {
  for {
    (cluster, label) <- baseClusterQuery.joinLeft(…).joinLeft(…)
  } yield (cluster, label)
}
// then:
clusterLabelRuntimeConfigQuery(clusterQuery.filter(_._1 === "foo").filter(_._._2 === "bar")).result
```
See also [this code](https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/ClusterComponent.scala#L262) which does something similar.